core/src/banking_stage.rs

```diff
@@ -1190,6 +1216,71 @@ impl BankingStage {
             .collect()
     }
 
+    fn generate_vote_packet_indexes(vers: &PinnedVec<Packet>) -> Vec<usize> {
+        vers.iter()
+            .enumerate()
+            .filter_map(|(index, ver)| {
+                if !ver.meta.discard {
+                    let tx: Transaction = limited_deserialize(&ver.data[0..ver.meta.size]).ok()?;
+                    if is_simple_vote_transaction(&tx) {
+                        Some(index)
+                    } else {
+                        None
+                    }
+                } else {
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn process_packets_vote(
+        mms: &mut Vec<Packets>,
+        vote_tx_count: &mut usize,
+        skip_process_packets_vote_count: &mut usize,
+        unprocessed_vote_tx_count: &mut usize,
+        poh: &Arc<Mutex<PohRecorder>>,
+        recorder: &TransactionRecorder,
+        transaction_status_sender: Option<TransactionStatusSender>,
+        gossip_vote_sender: &ReplayVoteSender,
+        banking_stage_stats: &BankingStageStats,
+    ) {
+        let mms_iter = mms.iter_mut();
+        for msgs in mms_iter {
+            let bank_start = poh.lock().unwrap().bank_start();
```

Comment:
I think it's safe to move this lock above outside of the `mms_iter` loop.  If we hit the end of a slot, we can then early exit and have the fallback logic handle it

Reply:
yep, good call!